core/test/matrix/csr.cpp

```diff
@@ -106,7 +106,6 @@ class Csr : public ::testing::Test {
         ASSERT_EQ(m->get_num_stored_elements(), 0);
         ASSERT_EQ(m->get_const_values(), nullptr);
         ASSERT_EQ(m->get_const_col_idxs(), nullptr);
-        ASSERT_EQ(m->get_const_row_ptrs(), nullptr);
```

Comment:
Since you always want to allocate for empty matrices, I think it would be best to also test for that:
```
ASSERT_NE(m->get_const_row_ptrs(), nullptr);
```